design/ceph/object-store-realm.md

```diff
@@ -0,0 +1,169 @@
+```
+
+#### Steps
+
+1. At this point the Rook operator recognizes that a new object store realm resource needs to be configured. The operator will start creating the resource to start the object store realm.
+
+2. After these steps the admin should start up a master [object-store-zonegroup](/design/ceph/object-store-zonegroup.md), a master [object-store-zone](/design/ceph/object-store-zone.md), and a [object-store](/design/ceph/object-store.md) in the master object-store-zone resource.
+
+3. Once all of the resources from step #2 are created, the operator will create a realm on the Rook Ceph cluster and the object-realm-resource resource will be running.
+
+#### Notes
+
+1. If a realm has already been created or pulled with the same name on the cluster, creation of this resource will fail.
+
+### Pulling the Realm from another Ceph Cluster
+
+#### Config
+
+When the storage admin is ready to sync data from another Ceph cluster with multisite set up, the first step is to pull the realm with the data from that cluster.
+
+To do this, the storage admin needs to create an object-store-realm with the same name as the realm being pulled from. This resource must have the `pullEndpoint` section configured. 
+
+The endpoint in the `pullEndpoint` section is an endpoint of an object-store in the master-zone. This endpoint must be resolvable.
+
+This example `object-store-realm.yaml`, pulls `my-realm` from a Ceph cluster with endpoint http://my-realm-zone-b-objectstore-1:80 in its master zone.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStoreRealm
+metadata:
+  name: my-realm
+  namespace: rook-ceph-2
+spec:
+  pullEndpoint: http://my-realm-zone-b-objectstore-1:80
+```
+
+Now create the object store realm.
+```bash
+kubectl create -f object-store-realm.yaml
+```
+
+#### Steps
+1. At this point the Rook operator recognizes that a new object store realm resource needs to be configured. The operator will start creating the resource to start the object store realm.
+
+2. After these steps the admin should start up a [object-store-zonegroup](/design/ceph/object-store-zonegroup.md). This zonegroup should be the same zonegroup of the zone the endpoint for in the pullRealm section is in.
+
+3. After the object-store-zonegroup resource is started, a non-master [object-store-zone](/design/ceph/object-store-zone.md), and a [object-store](/design/ceph/object-store.md) in the object-store-zone resource need to be started up.
+
+3. Once all of the resources steps #2 & #3 are created, the operator will pull the realm on the rook Ceph cluster and the resource will be running.
+
+#### Notes
+1. The endpoint in pullEndpoint must be resolvable from the cluster that is pulling the realm.
+2. The realm must have the same name in the `metadata` section as the realm it is pulling from the multisite enabled Ceph cluster.
+3. If a realm has already been created or pulled with the same name on the cluster, creation of this resource will fail.
```

Comment:
see my previous comment on this - don't think it should fail (unless it is different, and we dont support updates)